flake.nix

```diff
@@ -176,9 +176,11 @@
           export LIBSSH2_SYS_USE_PKG_CONFIG=1
 
           export RUSTFLAGS="-Zthreads=0"
-        '' + pkgs.lib.optionalString useMoldLinker ''
+        '' + (if pkgs.stdenv.isLinux then ''
           export RUSTFLAGS+=" -C link-arg=-fuse-ld=mold -C link-arg=-Wl,--compress-debug-sections=zstd"
-        '' + darwinNextestHack;
+        '' else ''
+          export RUSTFLAGS+=" -C link-arg=-fuse-ld=/usr/bin/ld -C link-arg=-ld_new"
+        '') + darwinNextestHack;
```

Comment:
By the way, it’s not directly related to the intent of this PR (so I’m not expecting it to change in this one either), but apparently the relevant `cargo-nextest` bug here has been fixed by now?

Reply:
Yep, looks like it. I'll file a separate PR.

Reply:
Actually, I'll just go ahead and put it in this PR in a separate commit. Will help avoid the (admittedly trivial) rebase conflicts.